src/base/TypeChecker.ml

```diff
@@ -798,6 +798,36 @@ module ScillaTypechecker (SR : Rep) (ER : Rep) = struct
             @@ add_stmt_to_stmts_env_gas
                  (TypedSyntax.CallProc (p, typed_args), rep)
                  checked_stmts remaining_gas
+        | ListIter (l, p) -> (
+            let%bind lt =
+              mark_error_as_type_error remaining_gas
+              @@ TEnv.resolveT env.pure (get_id l) ~lopt:(Some (get_rep l))
+            in
+            let l_type = rr_typ lt in
+            match
+              List.Assoc.find env.procedures ~equal:String.( = ) (get_id p)
```

Comment:
This line duplicates the one from `CallProc`, so we should probably separate it to a helper function.